doc/manual/src/language/string-interpolation.md

```diff
@@ -0,0 +1,82 @@
+# String interpolation
+
+String interpolation is a language feature where a [string], [path], or [attribute name] can contain expressions enclosed in `${ }` (dollar-sign with curly brackets).
+
+Such a string is an *interpolated string*, and an expression inside is an *interpolated expression*.
+
+Interpolated expressions must evaluate to one of the following:
+
+- [string]
+- [path]
+- [derivation]
```

Comment:
Counter-proposal to [this](https://github.com/NixOS/nix/pull/7278/files#r1039732574):
```suggestion
- a [string]
- a [path]
- a [derivation]
```

Link titles should reflect the contents, and in this case we're linking to the definition of the type, not a particular instance.